id_test.go

```diff
@@ -196,3 +196,25 @@ func BenchmarkFromString(b *testing.B) {
 // 		}
 // 	})
 // }
+
+func TestID_IsNil(t *testing.T) {
+	tests := []struct {
+		name string
+		id   ID
+		want bool
+	}{
+		{
+			name: "ID not nil",
+			id:   New(),
+			want: false,
+		},
+		{
+			name: "Nil ID",
+			id:   nilID,
```

Comment:
To make this test more realistic, you might want to use `ID{}` instead of `nilID`. It's more future proof.